go/vt/vtctl/workflow/utils.go

```diff
@@ -1019,3 +1021,59 @@ func applyTargetShards(ts *trafficSwitcher, targetShards []string) error {
 	}
 	return nil
 }
+
+// validateEmptyTables checks if all specified tables are empty across specified shards.
+// It queries each shard's primary tablet and if any non-empty table is found, it returns an error
+// containing a list of non-empty tables.
+func validateEmptyTables(ctx context.Context, ts *topo.Server, tmc tmclient.TabletManagerClient, shards []*topo.ShardInfo, tableSettings []*vtctldatapb.TableMaterializeSettings) error {
+	var mu sync.Mutex
+	isNonEmptyTable := map[string]bool{}
+
+	err := forAllShards(shards, func(shard *topo.ShardInfo) error {
+		primary := shard.PrimaryAlias
+		if primary == nil {
+			return fmt.Errorf("shard does not have a primary: %v", shard.ShardName())
+		}
+
+		ti, err := ts.GetTablet(ctx, primary)
+		if err != nil {
+			return err
+		}
+
+		eg, groupCtx := errgroup.WithContext(ctx)
+		eg.SetLimit(20)
+
+		for _, ts := range tableSettings {
```

Comment:
It might be worth adding the shard and table to the errors in this loop as there could be 1,000 shards and you would want to know what specific shard had a table with data in it, and e.g. what shard and table we encountered an error in processing.

Reply:
can we do this in a follow-up PR?